app.js

```diff
@@ -34,6 +35,7 @@ function clientLoaded (client) {
   client.on('StasisStart', function(event, channel) {
     if (!isDialed(event.args[0])) {
       var extension = event.args[0];
+      var confFile = process.argv[2];
       sla(client, confFile, channel, extension)
```

Comment:
Somewhere in your SLA function I would expect to see the HOLD_INTERCEPT function be applied to the channel. I haven't found the application of that dialplan function on a channel in this review; as it is, I'm not sure how you're intercepting hold events without it.